### PR TITLE
ADX 309 Improve DHIS2 Credentials UX

### PR DESCRIPTION
**Pull request by jonathansberry**
image

### CODE REVIEW

File: ckanext/dhis2harvester/templates/source/dhis2_credentials.html [hunk 1]
```diff
@@ -8,18 +9,44 @@ <H2>
...
+
 {% if data.dhis2_auth_token %}
-  {{ form.info(text=_('Auth token saved')) }}
+  {{form.info(text=_('A previous DHIS2 session is still active.  No need to log in again. '))}}
+  <div class="advanced-config toggle-box dhis2-login">
+  <div class="header">
+  <a href="#" class="toggle-btn toggled" data-module="toggle_box">{{_('Click here to change DHIS2 User')}}</a>
+  </div>
+  <div class="content" style="display:none">
+    {% call form.input("dhis2_username", id="field-dhis2-username", label=_("DHIS2 Username"), placeholder=_("dhis2_username"), value=data.dhis2_username, error=errors.dhis2_username, classes=["control-medium"], is_required=True) %}
```
Comment (tomeksabala) [inline ckanext/dhis2harvester/templates/source/dhis2_credentials.html:38]: Is it possible to remove this duplication of fields "dhis2_username" and "dhis2_password" from `{% if data.dhis2_auth_token %}` and `{% else %}` statements?
You're repeating them twice.

File: ckanext/dhis2harvester/templates/source/dhis2_credentials.html [hunk 2]
```diff
@@ -8,18 +9,44 @@ <H2>
...
+    {% endcall %}
+  </div>
+</div>
+
 {% if data.dhis2_auth_token %}
-  {{ form.info(text=_('Auth token saved')) }}
+  {{form.info(text=_('A previous DHIS2 session is still active.  No need to log in again. '))}}
+  <div class="advanced-config toggle-box dhis2-login">
+  <div class="header">
+  <a href="#" class="toggle-btn toggled" data-module="toggle_box">{{_('Click here to change DHIS2 User')}}</a>
```
Comment (tomeksabala) [inline ckanext/dhis2harvester/templates/source/dhis2_credentials.html:35]: Could you make clicking on the toggle link clear content of `dhis2_auth_token`? If the token is present it's processed with higher priority in the blueprint.
Other option is that we'd need to pass "update credentials" flag to the blueprint which sounds more complicated IMO.

Comment (jonathansberry) [reply ckanext/dhis2harvester/templates/source/dhis2_credentials.html:35]: So you're just suggesting {% set dhis2_auth_token = "" %}?

Comment (jonathansberry) [reply ckanext/dhis2harvester/templates/source/dhis2_credentials.html:35]: @tomeksabala Will change the order of precedence in the back end to make this work. I'll leve the PR open until this is done.

File: ckanext/dhis2harvester/templates/source/dhis2_credentials.html (hunk 1 above)
Comment (jonathansberry) [reply ckanext/dhis2harvester/templates/source/dhis2_credentials.html:38]: I'll do this with a macro

**Pull request merged**